### PR TITLE
CORTX-29889: add consul alive check for FAST FAILURE in rgw miniprovisioning

### PR DESCRIPTION
**Pull request by rohit-k-dwivedi**
Signed-off-by: Rohit Dwivedi <rohit.k.dwivedi@seagate.com>

# Problem Statement
- default connection check for consul takes almost 2 minutes of timeout, adding 3 retries with exponential backoff logic adds 7 sec + 4x60sec ie almost 8min 7 sec

# Design
- introduced check for consul using socket which checks if the server is alive with a timeout of 2 sec and 4 re-tries
  creates a fast fail which fails within 15sec .
# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [x] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: src/rgw/setup/rgw.py [hunk 1]
```diff
@@ -92,6 +94,7 @@ def config(conf: MappedConf):
         """Performs configurations."""
 
         Log.info('Config phase started.')
+        Rgw._is_consul_alive(conf)
```
Comment (cdeshmukh) [inline src/rgw/setup/rgw.py:97]: function name is misleading. we can rename it to _check_consul_connection()

File: src/rgw/setup/rgw.py [hunk 2]
```diff
@@ -215,6 +218,18 @@ def _get_consul_url(conf: MappedConf, seq: int = 0):
...
+    @ExponentialBackoff(exception=Exception, tries=4)
+    def _is_consul_alive(conf: MappedConf):
+        import socket
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(2)
+        consul_url = Rgw._get_consul_url(conf).split("//")[-1]
+        host, port = consul_url.split(':')
+        result = sock.connect_ex((host, port))
+        if result != 0:
+            raise SetupError(errno.EINVAL, "Consul server not alive")
```
Comment (cdeshmukh) [inline src/rgw/setup/rgw.py:231]: And message can be 
""Consul server not reachable through {ip:port}"

File: src/rgw/setup/rgw.py (hunk 1 above)
Comment (rohit-k-dwivedi) [reply src/rgw/setup/rgw.py]: done

File: src/rgw/setup/rgw.py (hunk 2 above)
Comment (rohit-k-dwivedi) [reply src/rgw/setup/rgw.py]: done

**Review by cdeshmukh — APPROVED** [review]

**Review by selvakumaar5496 — APPROVED** [review]

**Review by rahul-tripathi-git — APPROVED** [review]

**Pull request merged**